### PR TITLE
[RLlib] Torch DQN (APEX) TD-Error/prio. replay fixes.

### PR DESCRIPTION
**Pull request by sven1977**
PyTorch APEX_DQN with Prioritized Replay enabled would not work properly due to the td_error not being retrievable by the AsyncReplayOptimizer.

## Checks

- [x] I've run `scripts/format.sh` to lint the changes in this PR.
- [x] I've included any doc changes needed for https://docs.ray.io/en/latest/.
- [x] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failure rates at https://ray-travis-tracker.herokuapp.com/.
- Testing Strategy
   - [x] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested (please justify below)

### CODE REVIEW

File: rllib/agents/dqn/dqn_torch_policy.py [hunk 1]
```diff
@@ -244,6 +244,14 @@ def compute_q_values(policy, model, obs, explore, is_training=False):
     return q_values
 
 
+def grad_process_and_td_error_fn(policy, optimizer, loss):
```
Comment (sven1977) [inline rllib/agents/dqn/dqn_torch_policy.py:247]: Added td_error key to info dict (needed by AsyncReplayOptim to update PR weights).

File: rllib/agents/dqn/tests/test_apex.py [hunk 2]
```diff
@@ -14,19 +14,29 @@ def setUp(self):
     def tearDown(self):
         ray.shutdown()
 
-    def test_apex_epsilon_distribution(self):
+    def test_apex_compilation_and_per_worker_epsilon_values(self):
```
Comment (sven1977) [inline rllib/agents/dqn/tests/test_apex.py:17]: Merged test cases for APEX compilation (+ train() calls) and epsilon value (per-worker) checks.

File: rllib/tuned_examples/atari-apex.yaml [hunk 3]
```diff
@@ -22,7 +22,7 @@ apex:
           epsilon_timesteps: 200000
         prioritized_replay_alpha: 0.5
         final_prioritized_replay_beta: 1.0
-        final_prioritized_replay_beta_annealing_timesteps: 2000000
+        prioritized_replay_beta_annealing_timesteps: 2000000
```
Comment (sven1977) [inline rllib/tuned_examples/atari-apex.yaml:25]: Fixed unknown config key.

File: rllib/agents/dqn/tests/test_apex.py (hunk 2 above)
Comment (sven1977) [reply rllib/agents/dqn/tests/test_apex.py:17]: to save some time.

**Review by ericl — APPROVED** [review]
Looks good. hopefully compatible with https://github.com/ray-project/ray/pull/8077

**Pull request merged**